### PR TITLE
Fix a new production XML data wrinkle

### PR DESCRIPTION
**Pull request by mjgiarlo**
Namely, multiple consecutive whitespace characters in the middle of a tag

## Why was this change made?

To fix a production issue.

## Was the API documentation (openapi.yml) updated?

No.

## Does this change affect how this application integrates with other services?

No.

### CODE REVIEW

File: app/services/administrative_tags.rb [hunk 1]
```diff
@@ -131,7 +131,7 @@ def populate_database_tags_from_datastream!
         # There are a bunch of tags in production WITHOUT the padding spaces.
         # Fix that here unless already valid.
         tag_string.gsub!(':', ' : ') unless AdministrativeTag::VALID_TAG_PATTERN.match?(tag_string)
-        tag_string.strip!
+        tag_string.squish! # remove leading and trailing whitespace and remove double/triple/etc. spaces throughout
```
Comment (jmartin-sul) [inline app/services/administrative_tags.rb:134]: nice, didn't know about this method!

Comment (ndushay) [reply app/services/administrative_tags.rb:134]: it's ... squishy!

File: spec/services/administrative_tags_spec.rb [hunk 2]
```diff
@@ -88,10 +88,13 @@
         LegacyTagService.add(item_without_db_tags, 'One : Two : Three')
         LegacyTagService.add(item_without_db_tags, 'Two:Three:Four') # test weirdo tags
         LegacyTagService.add(item_without_db_tags, ' Three:Four:Five') # test weirdo tags
+        LegacyTagService.add(item_without_db_tags, 'Four :  Five :  Six') # test weirdo tags
       end
 
       it 'returns administrative tags from identity metadata XML' do
-        expect(described_class.for(item: item_without_db_tags)).to eq(['One : Two : Three', 'Two : Three : Four', 'Three : Four : Five'])
+        expect(described_class.for(item: item_without_db_tags)).to eq(
+          ['One : Two : Three', 'Two : Three : Four', 'Three : Four : Five', 'Four : Five : Six']
```
Comment (ndushay) [inline spec/services/administrative_tags_spec.rb:96]: if this spec is touched again, i'd like to see this array with a value on each line - it's hard to parse visually.
